tests/postgres_tests/test_indexes.py

```diff
@@ -18,10 +19,16 @@
 
 from . import PostgreSQLSimpleTestCase, PostgreSQLTestCase
 from .fields import SearchVector, SearchVectorField
+from .indexes import DummyIndex
 from .models import CharFieldModel, IntegerArrayModel, Scene, TextFieldModel
 
 
-class IndexTestMixin:
+class IndexTestMixin(ABC):
+    @property
+    @abstractmethod
+    def index_class(self):
+        pass
+
```

Comment:
Do we need this change? it looks unnecessary.

Reply:
`index_class` is an abstract property, but that's not documented and not obvious unless one studies `IndexTestMixin`'s methods. This just tells (a) the developer "hey, all subclasses of `IndexTestMixin` are expected to declare an `index_class` attribute" and (b) Python to raise when one subclasses `IndexTestMixin` without the `index_class` attribute:

```python
from abc import ABC, abstractmethod

class P(ABC):
    @property
    @abstractmethod
    def foo(self):
        ...

    def do_something(self):
        self.foo

class C(P):
    ...

C()  # -> TypeError: Can't instantiate abstract class C with abstract method foo
```

Python used to have [@abc.abstractproperty](https://docs.python.org/3/library/abc.html#abc.abstractproperty) but it was deprecated in favor of `@property` + `@abc.abstractmethod`.

Reply:
It's an internal test mixin and subclasses are not instantiated by developers per se, so I don't see much value in changing this. Nevertheless, if you strongly believe that's valuable, I'd prefer to add:
```python
class IndexTestMixin:
    @property
    def index_class(self):
        raise NotImplementedError("index_class must be set by a subclass.")
```
but that's a separate cleanup not related with this patch, so we can process it after merging this patch.

Reply:
`@property` + `@abc.abstractmethod` is a shortcut (and the standard method) for what you're suggesting. While I understand your being cautious, such hints help contributors to understand and navigate the codebase, esp. the newcomers.

Ultimately it's your decision.